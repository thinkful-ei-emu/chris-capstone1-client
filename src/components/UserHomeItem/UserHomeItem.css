.UserHomeItem {
    background-color: #CC8B56;
    display: flex;
    flex-direction: column;
    border: 0;
    text-decoration: none;
    box-shadow: 0px 11px 21px -8px rgba(0, 0, 0, 0.5);
    margin-bottom: 40px;
    margin-right: auto;
    margin-left: auto;
    border-radius: 10px;
    padding: 5px
  }
  
  .UserHomeItem:hover {
    cursor: pointer;
  }
  
  .UserHomeItem_info,
  .UserHomeItem_info:visited {
    color: black;
  }
  
  .UserHomeItem_image {
    height: 256px;
    width: 160px;
    border-radius: 10px 10px 0 0;
    background-size: cover;
    background-position: 50% 50%;
  }
  
  .UserHomeItem_details {
    padding: 10px;
    display: flex;
    flex: 1;
    flex-direction: column;
    justify-content: space-between;
    width: 250px;
  }
  
  .UserHomeItem_details h2 {
    margin: 0;
  }
  
  .UserHomeItem_details span {
    display: block;
    text-align: center;
    opacity: 0.5;
  }
  