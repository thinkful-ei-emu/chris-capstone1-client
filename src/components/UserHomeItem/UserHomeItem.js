import React from 'react';
import { Link } from 'react-router-dom';
import { Button } from '../Utils/Utils';
import './UserHomeItem.css';

export default class UserHomeItem extends React.Component {
    render () {
        const {book, bookDelete, list} = this.props;
        return (
            <div className='UserHomeItem'>
                <Link to={`/books/${book.id}`} className='UserHomeItem_info'>
                {list !== undefined && book.image && <img className='UserHomeItem_image' src={book.image} alt='the cover of the book' />}
                <div className='UserHomeItem_details'>
                    <div className='UserHomeItem_text'>
                        <h3 className='UserHomeItem_title'>{book.title}</h3>
                        <p className='UserHomeItem_author'>by: {book.author}</p>
                        {list !== undefined && <p>Average User Rating: <span>{book.average_rating}%</span> {' '}{' '}
                            #ofRatings: <span>{book.number_of_ratings}</span></p>}
                    </div>
                </div>
                </Link>
                <div className='buttons'>
                <Link to={`/edit/${book.id}`}>
                    <Button>
                        Edit
                    </Button>
                </Link>
                {' '}
                <Button onClick={() => bookDelete(book.id)}>
                    delete
                </Button>
                </div>
        </div>
        )
    }
}