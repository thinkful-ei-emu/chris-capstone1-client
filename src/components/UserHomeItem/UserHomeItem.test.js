import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserHomeItem from './UserHomeItem';
import ReactDom from 'react-dom';
import { BrowserRouter } from 'react-router-dom';

describe('UserHomeItem component', () => {
  let book = {
    "id": 1,
    "title": "Disappearing Earth",
    "author": "Julia Phillips",
    "list": "wishlist",
    "date_created": "2019-08-09T16:06:49.108Z",
    "book_source": "Print",
    "image": "https://media.vanityfair.com/photos/5d1662c85f741a0008ed05e1/master/w_800%2Cc_limit/best-books-of-2019-Disappearing-Earth.jpg",
    "book_report": "",
    "rating": 13,
    "genre": "Fairytale",
    "recommended": false,
    "user": {
        "id": 4,
        "user_name": "s.smith",
        "full_name": "Sam Smith",
        "email": "Sam@gmail.com",
        "date_created": "2019-08-09T16:06:49.108Z"
    },
    "poster": {
        "poster_report": null
    },
    "number_of_ratings": 5,
    "average_rating": 69.8
}
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render(
          <BrowserRouter>
            <UserHomeItem key={book.id} book={book} />
          </BrowserRouter>,
          div
        )
        ReactDom.unmountComponentAtNode(div)
      })
    it('renderes UserHomeItem by default', () => {
        expect(toJson(shallow(<UserHomeItem key={book.id} book={book} />))).toMatchSnapshot() 
    })
})