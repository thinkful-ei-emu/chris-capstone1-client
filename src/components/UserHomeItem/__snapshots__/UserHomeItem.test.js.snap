// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserHomeItem component renderes UserHomeItem by default 1`] = `
<div
  className="UserHomeItem"
>
  <Link
    className="UserHomeItem_info"
    to="/books/1"
  >
    <img
      alt="the cover of the book"
      className="UserHomeItem_image"
      src="https://media.vanityfair.com/photos/5d1662c85f741a0008ed05e1/master/w_800%2Cc_limit/best-books-of-2019-Disappearing-Earth.jpg"
    />
    <div
      className="UserHomeItem_details"
    >
      <div
        className="UserHomeItem_text"
      >
        <h3
          className="UserHomeItem_title"
        >
          Disappearing Earth
        </h3>
        <p
          className="UserHomeItem_author"
        >
          by: 
          Julia Phillips
        </p>
        <p>
          Average User Rating: 
          <span>
            69.8
            %
          </span>
           
           
           
          #ofRatings: 
          <span>
            5
          </span>
        </p>
      </div>
    </div>
  </Link>
  <div
    className="buttons"
  >
    <Link
      to="/edit/1"
    >
      <Button>
        Edit
      </Button>
    </Link>
     
    <Button
      onClick={[Function]}
    >
      delete
    </Button>
  </div>
</div>
`;
